[WebProfilerBundle] Show `APP_RUNTIME`

mudassaralichouhan opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 7.4
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Issues        | Fix #61491
| License       | MIT

This PR adds support for displaying the `APP_RUNTIME` value in the Symfony Profiler configuration dashboard.

**Before**
- The configuration dashboard did not show which application runtime was used.

**After**
- A new section "Application Runtime" is displayed when `$_SERVER['APP_RUNTIME']` is set, making it easier to debug and confirm the active runtime.

```php
$_SERVER['APP_RUNTIME'] = 'Symfony\\Runtime\\SymfonyRuntime';
```

**Profiler UI**
```
Application Runtime
-------------------
Runtime: Symfony\Runtime\SymfonyRuntime
```

mudassaralichouhan committed:
feature: Add APP_RUNTIME display in Symfony Profiler configuration dashboard

OskarStark commented on src/Symfony/Component/HttpKernel/DataCollector/ConfigDataCollector.php:
```diff
@@ -249,6 +250,14 @@ public function getSapiName(): string
     }
 
+    /**
+     * Gets the application runtime.
```
Can be removed

OskarStark commented on src/Symfony/Component/HttpKernel/Tests/DataCollector/ConfigDataCollectorTest.php:
```diff
@@ -77,6 +77,23 @@ public function testCollectWithoutKernel()
+    {
+        $c = new ConfigDataCollector();
+        
+        // Test without APP_RUNTIME set
```
Comments can be removed in this testcase

mudassaralichouhan committed:
profiler: display resolved application runtime class in Config panel

- Resolve runtime from APP_RUNTIME, composer.json extra.runtime.class, or Symfony\Runtime\SymfonyRuntime
- Expose via ConfigDataCollector::getAppRuntime and render in config.html.twig
- Add tests for env/composer/fallback resolution

mudassaralichouhan committed:
profiler: fix runtime resolution caching in collector tests

- Remove static cache from resolveAppRuntime() to avoid cross-test contamination

nicolas-grekas commented:
About the implementation, this relies too much on implementation details.
A better implementation my be to rely on the global `$runtime` variable.

BUT, I'm not sure what this info could be useful for. Any details about that?

The runner might be a more useful thing to show, because then one can see if there's running frankenphp or etc.
And this can be known by checking the first frames of any stack trace.

nicolas-grekas commented:
Closing for history. PR welcome I suppose to show the runner instead.
Thanks for submitting!
